File: src/news.js
sealed class Base {
   a0;
   constructor(a0) { this.a0 = 0; }
}

sealed class Sub1 extends Base {
   a1;
}

sealed class Sub2 extends Sub1 {
   a2;
   /*#@src3#*/constructor/*#/@src3#*/(a0, a1, a2) {
      super(a0);
      this.a1 = a1; this.a2 = a2;
   }
}

/*#@src#*/new Sub2(0, 1, 2)/*#/@src#*/;

File: src/news.c
void Sub2_CTOR(Sub2 *this, jsobj *a0, jsobj *a1, jsobj *a2) {
   { // super base class CTOR inlining
      Sub1 *super = (Sub1 *)this;
      *(&(super->inline0) + 0 * sizeof(jsobj *)) = a0;
   } // CTOR body
   *(&(this->inline0) + 1 * sizeof(jsobj *)) = a1;
   *(&(this->inline0) + 2 * sizeof(jsobj *)) = a2;
}


jsobj *o = GC_malloc(sizeof(jsobj) + 3 * sizeof(jsobj *));

o->__proto__ = Sub2_PROTOTYPE;
o->cmap = Sub2_HIDDENCLASS;
o->elements = 0L;
o->length = 3;

Sub2_CTOR(o, 0, 1, 2);
